decentraland/unity-explorer
# Feat: added sidebar

## What does this PR change?

Does this change the fate of the Universe? Probably not, but for sure it changes how we layout the main UI in DCL 💪.

Before we would have 4 separate UI elements with the Minimap, Chat, Persistent Explore Panel Opener and Persistent Emote Wheel Opener, each instantiated independently. This PR introduces a new UI element which houses the other ones, properly layouted, so the new UI element added (the sidebar) can interact properly with the rest. 
This means that during initialization, we instantiate the mainUIView which contain references to the other UI elements and then in each Plugin, we take those references to initialize their controllers.

The new sidebar has a special functionality that expands it when the mouse is near the left border of the screen, so code for that was added along with a new bus to disable hiding in certain cases.
The sidebar includes functionality to open the explore panel in specific tabs, as well as to display a profile mini-window and the new notifications view as well. These are shown on top of existing UI.

## How to test this beauty?

Just enter the game. You should see the sidebar by moving the mouse to the left side and it will softly move the other elements to make space for itself. When moving the mouse away from the sidebar, it will close back again into the nothingness. To keep it from hiding, on the top of the sidebar there is a (...) button which has a toggle to make the sidebar permanent for the session.
Please make sure all buttons work as expected, the visual state for everything should follow this figma doc -> https://www.figma.com/design/1WOzC6Qs27tkme1dBIFrSd/%F0%9F%97%BA%EF%B8%8F-Mini-Games-Ex-Quests-%7C-Explorer-2.0-%7C-07%2F24?node-id=6-3992&t=ShmhkpIuDd4p8Lg0-0

Have fun!
